src/server/json_family.cc

```diff
@@ -259,8 +279,108 @@ OpResult<vector<OptBool>> OpToggle(const OpArgs& op_args, string_view key, strin
   return vec;
 }
 
+template <typename Op>
+OpResult<string> OpDoubleArithmetic(const OpArgs& op_args, string_view key, string_view path,
```

Comment:
in general, I try to avoid templates unless it's absolutely necessary.
For example, here the performance benefit is negligible 
but templates introduce binary code bloating as well as less readable compiler errors, IMHO.
Usually, in cases like these, I use `function<double(double, double)>` typedef and make it a run-time dispatcher. 
I am not saying to change it now, but please know that templates have its price.